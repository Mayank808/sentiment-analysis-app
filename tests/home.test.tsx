import { render, screen } from "@testing-library/react";
import { vi } from "vitest";
import Home from "../app/page";

vi.mock("@clerk/nextjs", () => {
  // Create an mockedFunctions object to match the functions we are importing from the @nextjs/clerk package in the ClerkComponent component.
  const mockedFunctions = {
    auth: () =>
      new Promise((resolve) => resolve({ userId: "test_user_id_vite" })),
    ClerkProvider: ({ children }) => <div>{children}</div>,
    useUser: () => ({
      isSignedIn: true,
      user: {
        id: "test_user_id_vite",
        fullName: "test_user vitest",
      },
    }),
  };

  return mockedFunctions;
});

vi.mock("next/font/google", () => {
  return {
    Inter: () => ({ className: "inter" }),
  };
});

test(`Testing Home page render`, async () => {
  const home = render(await Home());
  expect(
    screen.getByText("Journel and reflect on yourself over time.")
  ).toBeTruthy();
  expect(screen.getByText("Get started")).toBeTruthy();

  expect(home).toMatchSnapshot();
});
