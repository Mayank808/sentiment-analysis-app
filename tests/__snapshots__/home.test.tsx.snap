// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Home page render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-screen h-screen bg-black flex justify-center items-center text-white"
      >
        <div
          class="w-full max-w-[600px] mx-auto text-center sm:text-left"
        >
          <h1
            class="text-6xl"
          >
            ScribeMe
          </h1>
          <p
            class="text-2xl text-white/60"
          >
            Journel and reflect on yourself over time.
          </p>
          <div
            class="my-4"
          >
            <a
              href="/scribe"
            >
              <button
                class="bg-blue-600 px-4 py-2 rounded-lg text-xl"
              >
                Get started
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-screen h-screen bg-black flex justify-center items-center text-white"
    >
      <div
        class="w-full max-w-[600px] mx-auto text-center sm:text-left"
      >
        <h1
          class="text-6xl"
        >
          ScribeMe
        </h1>
        <p
          class="text-2xl text-white/60"
        >
          Journel and reflect on yourself over time.
        </p>
        <div
          class="my-4"
        >
          <a
            href="/scribe"
          >
            <button
              class="bg-blue-600 px-4 py-2 rounded-lg text-xl"
            >
              Get started
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
